runtime/databricks/automl_runtime/forecast/prophet/model.py

```diff
@@ -225,15 +249,22 @@ def model_predict(self, df: pd.DataFrame, horizon: int = None) -> pd.DataFrame:
         result_df["ts_id"] = str(df["ts_id"].iloc[0])
         return result_df[return_cols]
 
-    def predict(self, context: mlflow.pyfunc.model.PythonModelContext, input_df: pd.DataFrame) -> pd.DataFrame:
+    def predict(self, context: mlflow.pyfunc.model.PythonModelContext, model_input: pd.DataFrame) -> pd.DataFrame:
         """
         Predict API from mlflow.pyfunc.PythonModel
         :param context: A :class:`~PythonModelContext` instance containing artifacts that the model
                         can use to perform inference.
-        :param input_df: Input dataframe
+        :param model_input: Input dataframe
         :return: A pd.DataFrame with the forecast components.
         """
-        return self._predict_impl(input_df)
+        self._validate_cols(model_input, self._id_cols + [self._time_col])
+        test_df = model_input.copy()
+        test_df["ts_id"] = test_df[self._id_cols].agg('-'.join, axis=1)
+        test_df.rename(columns={self._time_col: "ds"}, inplace=True)
+        group_cols = ["ts_id"] + self._id_cols
```

Comment:
Why do we need to group by on both ts_id and the id_cols? I thought it'll be the same as just grouping by ts_id?

Reply:
It is mainly used for joining with the orignal dataframe.